### DISCUSSION TITLE
Security Definer question

### DISCUSSION THREAD

**Discussion by prescience-data** [supabase/supabase#3093] [Questions] [answered]
If a function is defined as `SECURITY DEFINER` does it still have access to `auth.uid()` the way a `SECURITY INOKER` function would? Or will `auth.uid()` operate differently?

**Comment by soedirgo** [comment, marked as answer]
That depends on the owner of the function, but assuming you created the function via the dashboard, in which case the owner is `supabase_admin`, it should have access to `auth.uid()`. `auth.uid()` should behave the same way, i.e. it should reflect which signed-in user is calling your function.
